diff --git a/node_modules/react-native-google-mobile-ads/android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsNativeModule.kt b/node_modules/react-native-google-mobile-ads/android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsNativeModule.kt
index 6e70fdd..31371a3 100644
--- a/node_modules/react-native-google-mobile-ads/android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsNativeModule.kt
+++ b/node_modules/react-native-google-mobile-ads/android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsNativeModule.kt
@@ -25,6 +25,7 @@ import com.facebook.react.bridge.ReadableMap
 import com.facebook.react.module.annotations.ReactModule
 import com.google.android.gms.ads.AdListener
 import com.google.android.gms.ads.AdLoader
+import com.google.android.gms.ads.LoadAdError
 import com.google.android.gms.ads.MediaAspectRatio
 import com.google.android.gms.ads.VideoController.VideoLifecycleCallbacks
 import com.google.android.gms.ads.VideoOptions
@@ -41,7 +42,7 @@ class ReactNativeGoogleMobileAdsNativeModule(
 
   @ReactMethod
   override fun load(adUnitId: String, requestOptions: ReadableMap, promise: Promise) {
-    val holder = NativeAdHolder(adUnitId, requestOptions)
+    val holder = NativeAdHolder(adUnitId, requestOptions, promise)
     holder.loadAd { nativeAd ->
       val responseId = nativeAd.responseInfo?.responseId ?: return@loadAd
       adHolders[responseId] = holder
@@ -97,7 +98,11 @@ class ReactNativeGoogleMobileAdsNativeModule(
     return adHolders[responseId]?.nativeAd
   }
 
-  private inner class NativeAdHolder(private val adUnitId: String, private val requestOptions: ReadableMap) {
+  private inner class NativeAdHolder(
+    private val adUnitId: String,
+    private val requestOptions: ReadableMap,
+    private val promise: Promise
+  ) {
     var nativeAd: NativeAd? = null
       private set
 
@@ -117,6 +122,12 @@ class ReactNativeGoogleMobileAdsNativeModule(
       override fun onAdClosed() {
         emitAdEvent("closed")
       }
+
+      override fun onAdFailedToLoad(loadAdError: LoadAdError) {
+        val codeAndMessage =
+          ReactNativeGoogleMobileAdsCommon.getCodeAndMessageFromAdError(loadAdError)
+        promise.reject(codeAndMessage[0], codeAndMessage[1])
+      }
     }
 
     private val videoLifecycleCallbacks: VideoLifecycleCallbacks = object : VideoLifecycleCallbacks() {
